specs/sharding/beacon-chain.md

```diff
@@ -221,7 +224,8 @@ class BeaconState(merge.BeaconState):
     blob_builder_balances: List[Gwei, BLOB_BUILDER_REGISTRY_LIMIT]
     # A ring buffer of the latest slots, with information per active shard.
     shard_buffer: Vector[List[ShardWork, MAX_SHARDS], SHARD_STATE_MEMORY_SLOTS]
-    shard_sample_price: uint64
+    # Each shard tracks a base-fee to continually re-adjust towards target data capacity
+    shard_sample_prices: List[Wei, MAX_SHARDS]
```

Comment:
anyway around the uint256?

Reply:
In some places, yes, by representing the fee for all samples. My findings were that's not the case here, or we lose precision in base fee per sample. There are only a few computations with these, it's not critical to performance.

Reply:
Switched to Gwei :tada: